src/bentoml/_internal/store.py

```diff
@@ -159,25 +160,40 @@ def get(self, tag: t.Union[Tag, str]) -> Item:
             )
 
     @contextmanager
-    def register(self, tag: t.Union[str, Tag]):
+    def register(self, tag: t.Union[str, Tag]) -> t.Generator[str, None, None]:
         _tag = Tag.from_taglike(tag)
 
         item_path = _tag.path()
-        if self._fs.exists(item_path):
-            raise BentoMLException(
+        if self.exists(_tag):
+            raise ItemAlreadyExists(
                 f"Item '{_tag}' already exists in the store {self._fs}"
             )
         self._fs.makedirs(item_path)
         try:
-            yield self._fs.getsyspath(item_path)
-        finally:
+            yield self.path_of(item_path)
+        except KeyboardInterrupt:
+            # During save, if user hit Ctrl+C, we will remove the item path cleanly
+            # so that validate() will work correctly.
+            self._fs.removetree(item_path)
```

Comment:
If I don't miss anything the clear up is only required for `use_temp_fs=False`, but the `register()` isn't on the path of `use_temp_fs=False` right?

It should be moved here instead: https://github.com/bentoml/BentoML/blob/c794024eef895405fbfe53dea5e9b129ed60b5b3/src/bentoml/models.py#L312-L313

Reply:
Fine, then moving to bentoml.models handles both.

Reply:
Oh you are correct, let me do that.